src/lib/ngc/ngcc-processor.ts

```diff
@@ -0,0 +1,114 @@
+import { Logger, PathMappings } from '@angular/compiler-cli/ngcc';
+import { existsSync } from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+import * as log from '../util/log';
+import { EntryPointNode, ngUrl } from '../ng-v5/nodes';
+
+// Transform a package and it's typings when NGTSC is resolving a module.
+export class NgccProcessor {
+  private _processedModules = new Set<string>();
+  private _logger: NgccLogger;
+  private _nodeModulesDirectory: string;
+  private _ngcc: typeof import('@angular/compiler-cli/ngcc');
+  private _pathMappings: PathMappings | undefined;
+  private _entryPointsUrl: string[];
+
+  constructor(private readonly compilerOptions: ts.CompilerOptions, private readonly entryPoints: EntryPointNode[]) {
+    this._ngcc = require('@angular/compiler-cli/ngcc');
+    this._entryPointsUrl = this.entryPoints.map(({ url }) => ngUrl(url));
+
+    const { baseUrl, paths } = this.compilerOptions;
+    this._nodeModulesDirectory = this.findNodeModulesDirectory(baseUrl);
+
+    if (baseUrl && paths) {
+      this._pathMappings = {
+        baseUrl,
+        paths,
+      };
+    }
+  }
+
+  processModule(moduleName: string, resolvedModule: ts.ResolvedModule | ts.ResolvedTypeReferenceDirective): void {
+    const resolvedFileName = resolvedModule.resolvedFileName;
+    if (
+      !resolvedFileName ||
+      moduleName.startsWith('.') ||
+      this._processedModules.has(moduleName) ||
+      this._entryPointsUrl.includes(ngUrl(moduleName))
+    ) {
+      // Skip when module is unknown, relative or NGCC compiler is not found or already processed.
```

Comment:
This comment looks inaccurate. There's no check for `... NGCC compiler is not found ...`

<sup>It's also inaccurate in `@ngtools/webpack` ;)</sup>

Reply:
Old comment 😜